src/main/java/net/datafaker/annotations/FakeResolver.java

```diff
@@ -0,0 +1,67 @@
+package net.datafaker.annotations;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Objects;
+
+import net.datafaker.transformations.JavaObjectTransformer;
+import net.datafaker.transformations.Schema;
+
+public class FakeResolver<T> {
+
+    private final Class<T> clazz;
+
+    public FakeResolver(Class<T> clazz) {
+        this.clazz = clazz;
+    }
+
+    public T get(String callingClassName, Schema<Object, ?> schema) {
+        checkFakeAnnotation(clazz);
+
+        Fake fakeAnnotation = clazz.getAnnotation(Fake.class);
+        Schema<Object, ?> useSchema = getSchema(schema, callingClassName, fakeAnnotation.value());
+
+        JavaObjectTransformer jTransformer = new JavaObjectTransformer();
+        return (T) jTransformer.apply(clazz, useSchema);
+    }
+
+    private static String[] getClassNameAndMethod(String callingClassName, String[] defaultSchema) {
+        if (defaultSchema.length != 0 && defaultSchema[0].contains("#")) {
+            return defaultSchema[0].split("#");
+        } else {
+            String[] classAndMethod = new String[2];
+            classAndMethod[0] = callingClassName;
+            classAndMethod[1] = defaultSchema[0];
+            return classAndMethod;
+        }
+    }
+
+    private Schema<Object, ?> getSchema(Schema<Object, ?> schema, String callingClassName, String[] pathToSchema) {
+        if (schema != null) {
+            return schema;
+        } else if (pathToSchema.length != 0) {
+            try {
+                String[] classNameAndMethod = getClassNameAndMethod(callingClassName, pathToSchema);
+                Class<?> myClass = Class.forName(classNameAndMethod[0]);
+                Method myStaticMethod = myClass.getMethod(classNameAndMethod[1]);
+                myStaticMethod.setAccessible(true);
+                return (Schema<Object, ?>) myStaticMethod.invoke(null);
```

Comment:
```suggestion
                String[] classNameAndMethod = getClassNameAndMethod(callingClassName, pathToSchema);
                Class<?> myClass = Class.forName(classNameAndMethod[0]);
                Method myStaticMethod = myClass.getMethod(classNameAndMethod[1]);
                myStaticMethod.setAccessible(true);
                return (Schema<Object, ?>) myStaticMethod.invoke(null);
```
final int sharpIndex = pathToSchema[0].indexOf('#');
                final Class<?> classToCall;
                final String methodName;
                if (sharpIndex >= 0) {
                    classToCall = Class.forName(pathToSchema[0].substring(0, sharpIndex));
                    methodName = pathToSchema[0].substring(sharpIndex + 1);
                } else {
                    classToCall = callingClass;
                    methodName = pathToSchema[0];
                }
                Method myStaticMethod = classToCall.getMethod(methodName);